Feature/allow duplicate names

disconsented opened the pull request:
Remove the unique constraint on server names

LordRalex committed:
Continue migration work

LordRalex committed:
Fix docker to use bytes as well

UI does the conversions, we don't need to here

LordRalex committed:
Move commands to be stored at root for all cases

LordRalex committed:
Attempt build fix

LordRalex committed:
Update build.yml

LordRalex committed:
Update build.yml

LordRalex committed:
Update build.yml

LordRalex committed:
Move docker to use /pufferpanel for the mountpoint
This makes it conistent and we don't require an old path to exist, and there won't be collision with the name

LordRalex committed:
Continue work on other stuff

LordRalex committed:
Fix compile

Diablillowilly committed:
register now indicates valid username conditions (#948)

LordRalex committed:
New Crowdin updates (#939)

Diablillowilly committed:
ScopeServersConsoleSend was not used in front and backend (#949)

LordRalex committed:
Documentation!

LordRalex committed:
Finish panel api documentation

LordRalex committed:
Move get check to be when it's needed
This now means you can technically "write" files without being able to read them

LordRalex committed:
Update servers.json

LordRalex committed:
New Crowdin updates (#954)

nepcore committed:
make the frontend console support ansi escape codes

LordRalex committed:
Fix bug where saving empty user perms deletes all perms
This is why ORMs suck

LordRalex committed:
When you don't feel like fixing the test, delete it

nepcore committed:
make user create/edit view work better

LordRalex committed:
Add TERM support since we have some colors

LordRalex committed:
Fix build

LordRalex committed:
New Crowdin updates (#956)

Diablillowilly committed:
pufferpanel user add now follows same validations as the user api (#947)

LordRalex committed:
Update release.yml

[skip]

LordRalex committed:
Update build.yml

LordRalex committed:
Update release.yml

LordRalex committed:
Update release.yml

LordRalex committed:
Update release.yml

LordRalex committed:
Update pty and go to resolve tty issues

nepcore committed:
redo input handling to be simpler and more reusable

LordRalex committed:
Revert docker using /pufferpanel
Because of how commands work, it's resolving where it thinks it should be at on the host level, but when in docker, it's not that path. This causes srcds to fail to work properly

ToonSpin committed:
Fix Swagger docs for server endpoints (#964)

nepcore committed:
expose environment options in a generic way (and make use of that for so far unexposed docker options)

nepcore committed:
made template editor less wonky and less annoying

LordRalex committed:
Add delete template endpoint

LordRalex committed:
Revert "Revert docker using /pufferpanel"

This reverts commit e660165ba6b58ded2bd66126e321c29fdd36750d.

nepcore committed:
add button to delete a template

nepcore committed:
rewire frontend api access, make websocket more error resitant and implement http fallback for ebsocket operations

nepcore committed:
add an alter file operation

nepcore committed:
update js dependencies

LordRalex committed:
Move tty to be default environment if on linux
TTY will be "deprecated" for the term standard, which will be usable on both windows and linux.

nepcore committed:
fix file manager sorting, fixes #981

LordRalex committed:
Add endpoint to get oauth scopes

LordRalex committed:
Fix swagger docs

LordRalex committed:
Update build.yml

LordRalex committed:
Fix build

thecmdradama committed:
Fix for users unable to view servers when using PostgreSQL. (#979)

nepcore committed:
template editor overhaul and server definition editor

nepcore committed:
rename see servers permission; fixes #925

nepcore committed:
make the import templates button more visible

nepcore committed:
don't rely on backend for sorting; fixes #985

nepcore committed:
properly url encode file names for use in urls; fixes #931

disconsented committed:
Merge pull request #1 from PufferPanel/master

Update from upstream

disconsented committed:
Merge pull request #1 from PufferPanel/master

Update from upstream

LordRalex committed:
Add missing options

LordRalex committed:
New Crowdin updates (#960)

LordRalex committed:
Start switching away from FPM

This will fail since RPM isn't in place yet

LordRalex committed:
Update build.yml

disconsented committed:
Expose resources and complete work for publishing ports (#994)

disconsented committed:
Mailjet (#995)

LordRalex committed:
Update build.yml

LordRalex committed:
Update build.yml

LordRalex committed:
Update build.yml

LordRalex committed:
Update build.yml

nepcore committed:
added json mode for the server definition editor

nepcore committed:
clean up useless debug logging...

LordRalex committed:
Yolo concurrency

LordRalex committed:
Update build.yml

LordRalex committed:
Update build.yml

LordRalex committed:
Update build.yml

LordRalex committed:
Update build.yml

LordRalex committed:
Update build.yml

LordRalex committed:
Update build.yml

LordRalex committed:
Update build.yml

LordRalex committed:
Update build.yml

LordRalex committed:
Update build.yml

LordRalex committed:
Attempt to add RPM builds

LordRalex committed:
Fix map deletion

disconsented committed:
Remove the unique index from Name, allowing for duplicate names

disconsented committed:
Remove the unique index from Name, allowing for duplicate names

nepcore committed:
fix sftp permission check

LordRalex committed:
Update release.yml

nepcore committed:
offer conversion when template editor sees a v1 template json

nepcore committed:
make invite links actually work

LordRalex committed:
Update build.yml

LordRalex committed:
Change scripts to handle errors a bit more correctly

LordRalex committed:
If an option request, don't require an oauth token since it does not leak data

nepcore committed:
unbreak server creation

LordRalex committed:
Update build.yml

LordRalex committed:
Fix bash commands

nepcore committed:
restart service after upgrade

nepcore committed:
Update build.yml

nepcore committed:
fix postupgrade failing when the service isn't running

disconsented committed:
Merge pull request #2 from PufferPanel/master

Syncing from upstream

disconsented committed:
Merge pull request #2 from PufferPanel/master

Syncing from upstream

LordRalex committed:
Update main.go

disconsented committed:
Allow for servers to be created with an ID (#1004)

LordRalex commented:
This doesn't apply retro-actively, so any installation before this would have the constraint while new ones don't. We would need a conversion.

disconsented commented:
A migration via gorm?

LordRalex committed:
Fix window builds

LordRalex committed:
Update build.yml

LordRalex committed:
Update release.yml

disconsented committed:
Feature/zip api (#986)

* Add Archive/Extract support
* Comment adjustment to fix CI
* Add extract/archive as operations
* Check the destination doesn't exist for extraction/archiving
* Check we have files before trying to archive them

LordRalex committed:
Update release.yml

LordRalex commented:
Yes. What should happen is that 2.2.0 installations should be able to migrate to this and support this.

dependabot[bot] committed:
Bump ansi_up from 4.0.4 to 5.0.0 in /client (#1002)

Bumps [ansi_up](https://github.com/drudru/ansi_up) from 4.0.4 to 5.0.0.
- [Release notes](https://github.com/drudru/ansi_up/releases)
- [Commits](https://github.com/drudru/ansi_up/compare/v4.0.4...v5.0.0)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

LordRalex committed:
New Crowdin updates (#989)

LordRalex committed:
Update release.yml

nepcore committed:
stop serving useless duplicate css

nepcore committed:
minor improvement for cpu chart

nepcore committed:
minor convenience features for server creator

nepcore committed:
console overhaul (visually distinct daemon messages, carriage return support, inline pausing)

nepcore committed:
sanitize unknown error details

nepcore committed:
lint cleanup

nepcore committed:
implement archive/extract functionality in file manager

nepcore committed:
added file type indicators to the file manager

nepcore committed:
disable scrolling while a modal is open

nepcore committed:
allow overriding configuration from database and expose settings on the panel

LordRalex committed:
Start a little more migration work

LordRalex committed:
Add "container" as a supported term
This will be used later when we support podman

disconsented commented:
Well, this is annoying.

Gorm doesn't really have any support for versioned migrations in this sense, so this expands the scope quite a bit.

Unless you have any strong opinions otherwise I don't see much value in going beyond something like:

- Table to track migrations probably structured like: version (i64), name (string), success (bool). Where version is just the epoch time of when it was created.
- Upon startup run through a sorted slice of the migrations, attempting to apply them, panic if one fails.

Since this will be done in code there is no sensible way to calculate a signature for integrity or provide a downgrade path.

Thoughts?

nepcore committed:
drag and drop file upload

nepcore committed:
fix disabling overlay background scroll behaving weirdly

nepcore committed:
visual fix for sftp info display

LordRalex committed:
Add working dir for main command

LordRalex committed:
Update forge maven url

LordRalex committed:
New Crowdin updates (#1013)

* New translations errors.json (German)

* New translations common.json (Chinese Simplified)

* New translations env.json (Chinese Simplified)

* New translations errors.json (Chinese Simplified)

* New translations files.json (Chinese Simplified)

* New translations nodes.json (Chinese Simplified)

* New translations scopes.json (Chinese Simplified)

* New translations servers.json (Chinese Simplified)

* New translations templates.json (Chinese Simplified)

* New translations users.json (Chinese Simplified)

* New translations common.json (French)

* New translations errors.json (French)

* New translations templates.json (French)

* New translations common.json (Korean)

* New translations env.json (French)

nepcore committed:
allow saving account specific settings in database and make theme selection and dark mode use that

nepcore committed:
minor cleanup

nepcore committed:
drag and drop server list reordering

LordRalex committed:
Fix config path for arm builds

nepcore committed:
expose working directory template option

nepcore committed:
change user settings value column to be of type text

Commit:
Add darwin/macos support

Commit:
Add darwin/macos support

Commit:
Add darwin/macos support round 2

Commit:
Add darwin/macos support round 2

disconsented committed:
Merge branch 'PufferPanel:master' into master

disconsented committed:
Merge branch 'PufferPanel:master' into master

nepcore committed:
don't throw errors on server list when permissions to a server are retracted

nepcore committed:
replace scroll heavy server page with a tab based alternative

nepcore committed:
slight ux improvement on the restart button

nepcore committed:
added 2fa support

LordRalex committed:
New Crowdin updates (#1018)

LordRalex commented:
I'm actually curious looking at this again, what is the use case for the same name on servers? How as a user is it good?

disconsented commented:
It lets use N users have a "Minecraft" server on the same panel.

More so, it means I don't need to enforce it for when I am provisioning servers. I'm not quite sure why there is a requirement for both a unique ID and name.

nepcore committed:
lint cleanup

nepcore committed:
prevent server list flickering caused by hardware acceleration

nepcore committed:
remove some derpiness in the template editor

LordRalex committed:
Escape panel-specific ENVs

LordRalex committed:
Move to building images ourselves, create types for deployments

LordRalex committed:
Fix workflow file

LordRalex committed:
Update build.yml

LordRalex committed:
Update build.yml

LordRalex committed:
Update build.yml

LordRalex committed:
Update build.yml

LordRalex committed:
Update build.yml

LordRalex committed:
Try to fix builds

LordRalex committed:
Javac not in path

LordRalex committed:
Switch massive to be latest and latest to base

LordRalex committed:
Fix database loading with config, clean up docker code for mounting

LordRalex committed:
Switch images to use base-devel

LordRalex committed:
Clean up java so it's usable via templates

LordRalex committed:
Base big image on ubuntu because of srcds...
SRCDS demands a 32bit lib.... but if you do it in alpine, you break java............

LordRalex committed:
Get containers to actually work right....

LordRalex committed:
Move settings to allow setting multiple at once, move to specific scope

nepcore committed:
save settings in batch instead of using one request per setting

LordRalex committed:
Change to use options for checking SSL

LordRalex committed:
Force key to exist

LordRalex committed:
Ensure oauth check supports failing

LordRalex committed:
Add config for rpm

LordRalex committed:
Update build.yml

LordRalex committed:
Update build.yml

LordRalex committed:
Use database from middleware where possible

LordRalex committed:
Add logging related to token services

LordRalex committed:
Fix timer on token

LordRalex committed:
Fix config to work with binary data on mysql

nepcore committed:
update npm package name in dockerfile

LordRalex committed:
New Crowdin updates (#1032)

disconsented committed:
Add macos support (round2) (#1027)

disconsented committed:
Feature/scheduled tasks (#1029)

LordRalex committed:
Downgrade back to ubuntu to resolve glibc errtors

LordRalex committed:
Update release.yml

disconsented committed:
Merge remote-tracking branch 'upstream/master'

# Conflicts:
#	environments/environmentloader_darwin.go

disconsented committed:
Merge remote-tracking branch 'upstream/master'

# Conflicts:
#	environments/environmentloader_darwin.go

disconsented committed:
Merge remote-tracking branch 'upstream/master' into feature/allow_duplicate_names

disconsented committed:
Merge remote-tracking branch 'upstream/master' into feature/allow_duplicate_names

disconsented committed:
Implemented gorm v2 && database migrations

disconsented committed:
Implemented gorm v2 && database migrations

disconsented committed:
Almost certainly broke migration but it 'works'

disconsented committed:
Almost certainly broke migration but it 'works'

disconsented committed:
Goofed up some of the operation names (#1041)

nepcore committed:
scheduled tasks first iteration (no it's not finished yet)

LordRalex committed:
Fix editing to completely replace

LordRalex committed:
Only add local node if no other nodes

LordRalex committed:
Add features endpoint

disconsented committed:
Updating types to clean up most of the uints
Moved ports to uint16

disconsented committed:
Updating types to clean up most of the uints
Moved ports to uint16

disconsented committed:
Fixed the scheduler causing errors if it wasn't active (#1045)

LordRalex commented on database/loader.go:
```diff
@@ -61,22 +70,45 @@ func openConnection() (err error) {
+		dialector = mysql.Open(connString)
+	case "postgresql":
+		dialector = postgres.Open(connString)
+	default:
```
missing SQL Server, which is supported by gorm

disconsented committed:
Add sqlserver

disconsented committed:
Add sqlserver

LordRalex commented on conversion.go:
```diff
@@ -58,7 +58,7 @@ func Convert(val interface{}, target interface{}) (interface{}, error) {
 			return uint(0), nil
 		}
-		return cast.ToUintE(val)
+		return cast.ToIntE(val)
```
This returns a uint, so this needs to be reverted.

LordRalex commented on database/loader.go:
```diff
@@ -14,16 +14,26 @@
 	"github.com/pufferpanel/pufferpanel/v2/logging"
 	"github.com/pufferpanel/pufferpanel/v2/models"
+	"gorm.io/driver/mysql"
+	_ "gorm.io/driver/mysql"
```
We can remove the _ references if it's already imported

LordRalex commented on models/node.go:
```diff
@@ -16,17 +16,18 @@ package models
 
 type Node struct {
-	ID          uint   `json:"-"`
+	ID          int    `json:"-"`
```
These should remain as uint

LordRalex commented on services/permission.go:
```diff
@@ -44,22 +44,22 @@ func (ps *Permission) GetForServer(serverId string) ([]*models.Permissions, erro
 	err := ps.DB.Preload("User").Preload("Server").Where(permissions).First(permissions).Error
 
-	if err != nil && gorm.IsRecordNotFoundError(err) {
+	if err != nil && gorm.ErrRecordNotFound == err {
```
https://gorm.io/docs/v2_release_note.html#ErrRecordNotFound says it should be done as a errors.Is

LordRalex commented on services/server.go:
```diff
@@ -90,41 +89,28 @@ func (ss *Server) Update(model *models.Server) error {
-	}
-
-	if inTrans {
+	trans := ss.DB
```
This call is interesting, because it actually looks to see if we are in a transaction already, and not create a new one if we are. We should keep this logic, unless the delete does the daemon call first.
This is because if the daemon doesn't delete the server, we have to roll the database back. This would always create a new one and commit it even if the daemon fails.

Romaindu35 committed:
Fix error on swagger (#1049)

Swagger say that the link for the api to get deployment is ​/api​/nodes​/{id}​/deploy but good link is ​/api​/nodes​/{id}​/deployment

LordRalex committed:
Have a flag for disabling registrations
Not used yet, only defined for UI to implement

LordRalex committed:
Pull from config for flag

LordRalex committed:
Disable reg feature if disabled
This will prevent sub-users....

LordRalex committed:
Update features to indicate if docker is actually running

LordRalex committed:
Move scheduler stop to be done before we delete the server

LordRalex committed:
Remove stop because docker does this already, clean up logging to console

nepcore committed:
don't try loading server data the user doesn't have access to

LordRalex committed:
Fully wrap the stops for the scheduler

LordRalex committed:
Fix toggles on admin section for server restarts

LordRalex committed:
Add extra libs for massive image

nepcore committed:
some console performance optimization

disconsented committed:
Remove redundant imports

disconsented committed:
Remove redundant imports

disconsented committed:
Revert ID's back to uints

disconsented committed:
Revert ID's back to uints

disconsented committed:
Move to using errors.is with gorm

disconsented committed:
Move to using errors.is with gorm

disconsented committed:
Fix the unit conversion

disconsented committed:
Fix the unit conversion

disconsented committed:
Cast the id

disconsented committed:
Cast the id

disconsented committed:
Merge remote-tracking branch 'origin/master' into feature/allow_duplicate_names

disconsented committed:
Merge remote-tracking branch 'origin/master' into feature/allow_duplicate_names

disconsented committed:
Merge remote-tracking branch 'upstream/master' into feature/allow_duplicate_names

# Conflicts:
#	config/config.go

disconsented committed:
Merge remote-tracking branch 'upstream/master' into feature/allow_duplicate_names

# Conflicts:
#	config/config.go

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/PufferPanel/PufferPanel?pullRequest=1006) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/PufferPanel/PufferPanel?pullRequest=1006) before we can accept your contribution.<br/>**1** out of **2** committers have signed the CLA.<br/><br/>:white_check_mark: disconsented<br/>:x: James Kerr<br/><hr/>**James Kerr** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/PufferPanel/PufferPanel?pullRequest=1006) it.</sub>

LordRalex committed:
New Crowdin updates (#1039)

LordRalex committed:
Update release.yml

LordRalex committed:
Update release.yml

LordRalex committed:
Add oauth2 endpoint for servers
Creating oauth2 clients for servers

LordRalex committed:
Update oauth tokens to use user/server perms if no explicit scopes defined

LordRalex committed:
Add oauth2 endpoints for self

nepcore committed:
user friendly task editor (hidden for now due to task system issues)

LordRalex committed:
Update docker.go

nepcore committed:
add a bunch of hotkeys

nepcore committed:
make the frontend respect the resistration disable flag

LordRalex committed:
Add some more emails

LordRalex committed:
More emails

LordRalex committed:
Add oauth2 api docs for account

LordRalex committed:
Add MSSQL to main binary
There is not a real benefit to limiting this to Windows builds, so we will just always include it

nepcore committed:
added oauth client management to account and server pages

nepcore committed:
significant performance improvement for the console

nepcore committed:
tiny little fix for a tiny little derp on the oauth interface

nepcore committed:
handle server creation errors properly

nepcore committed:
slightly improved node management

nepcore committed:
forgot to remove some debug code

disconsented committed:
Merge remote-tracking branch 'upstream/master' into feature/allow_duplicate_names

# Conflicts:
#	database/loader.go
#	database/loader_windows.go
#	go.mod
#	go.sum
#	models/client.go

disconsented committed:
Correct id's back to uints

disconsented committed:
Merge remote-tracking branch 'origin/feature/allow_duplicate_names' into feature/allow_duplicate_names

# Conflicts:
#	.github/workflows/release.yml
#	assets/email/emails.json
#	client/package-lock.json
#	client/package.json
#	client/src/App.vue
#	client/src/components/server/admin.vue
#	client/src/components/server/console.vue
#	client/src/components/server/controls.vue
#	client/src/components/server/files.vue
#	client/src/components/server/tasks.vue
#	client/src/components/server/type/generic.vue
#	client/src/lang/ar_SA/servers.json
#	client/src/lang/ar_SA/settings.json
#	client/src/lang/ar_SA/users.json
#	client/src/lang/cs_CZ/servers.json
#	client/src/lang/cs_CZ/settings.json
#	client/src/lang/cs_CZ/users.json
#	client/src/lang/da_DK/servers.json
#	client/src/lang/da_DK/settings.json
#	client/src/lang/da_DK/users.json
#	client/src/lang/de_DE/servers.json
#	client/src/lang/de_DE/settings.json
#	client/src/lang/de_DE/users.json
#	client/src/lang/el_GR/servers.json
#	client/src/lang/el_GR/settings.json
#	client/src/lang/el_GR/users.json
#	client/src/lang/en_US/common.json
#	client/src/lang/en_US/errors.json
#	client/src/lang/en_US/nodes.json
#	client/src/lang/en_US/servers.json
#	client/src/lang/en_US/settings.json
#	client/src/lang/es_419/servers.json
#	client/src/lang/es_419/settings.json
#	client/src/lang/es_419/templates.json
#	client/src/lang/es_419/users.json
#	client/src/lang/es_ES/files.json
#	client/src/lang/es_ES/servers.json
#	client/src/lang/es_ES/settings.json
#	client/src/lang/es_ES/users.json
#	client/src/lang/et_EE/servers.json
#	client/src/lang/et_EE/settings.json
#	client/src/lang/et_EE/users.json
#	client/src/lang/fi_FI/servers.json
#	client/src/lang/fi_FI/settings.json
#	client/src/lang/fi_FI/users.json
#	client/src/lang/fr_FR/servers.json
#	client/src/lang/fr_FR/users.json
#	client/src/lang/he_IL/servers.json
#	client/src/lang/he_IL/settings.json
#	client/src/lang/he_IL/users.json
#	client/src/lang/hu_HU/servers.json
#	client/src/lang/hu_HU/settings.json
#	client/src/lang/hu_HU/users.json
#	client/src/lang/id_ID/servers.json
#	client/src/lang/id_ID/settings.json
#	client/src/lang/id_ID/users.json
#	client/src/lang/it_IT/servers.json
#	client/src/lang/it_IT/settings.json
#	client/src/lang/it_IT/users.json
#	client/src/lang/ja_JP/servers.json
#	client/src/lang/ja_JP/settings.json
#	client/src/lang/ja_JP/users.json
#	client/src/lang/ko_KR/servers.json
#	client/src/lang/ko_KR/settings.json
#	client/src/lang/ko_KR/users.json
#	client/src/lang/nl_NL/servers.json
#	client/src/lang/nl_NL/settings.json
#	client/src/lang/nl_NL/users.json
#	client/src/lang/no_NO/servers.json
#	client/src/lang/no_NO/settings.json
#	client/src/lang/no_NO/users.json
#	client/src/lang/pl_PL/files.json
#	client/src/lang/pl_PL/servers.json
#	client/src/lang/pl_PL/settings.json
#	client/src/lang/pl_PL/users.json
#	client/src/lang/pt_BR/servers.json
#	client/src/lang/pt_BR/settings.json
#	client/src/lang/pt_BR/users.json
#	client/src/lang/pt_PT/servers.json
#	client/src/lang/pt_PT/settings.json
#	client/src/lang/pt_PT/users.json
#	client/src/lang/ro_RO/servers.json
#	client/src/lang/ro_RO/settings.json
#	client/src/lang/ro_RO/users.json
#	client/src/lang/ru_RU/servers.json
#	client/src/lang/ru_RU/settings.json
#	client/src/lang/ru_RU/templates.json
#	client/src/lang/ru_RU/users.json
#	client/src/lang/sr_SP/servers.json
#	client/src/lang/sr_SP/settings.json
#	client/src/lang/sr_SP/users.json
#	client/src/lang/sv_SE/servers.json
#	client/src/lang/sv_SE/settings.json
#	client/src/lang/sv_SE/users.json
#	client/src/lang/tr_TR/servers.json
#	client/src/lang/tr_TR/settings.json
#	client/src/lang/tr_TR/users.json
#	client/src/lang/uk_UA/servers.json
#	client/src/lang/uk_UA/settings.json
#	client/src/lang/uk_UA/users.json
#	client/src/lang/vi_VN/servers.json
#	client/src/lang/vi_VN/settings.json
#	client/src/lang/vi_VN/users.json
#	client/src/lang/zh_CN/servers.json
#	client/src/lang/zh_CN/settings.json
#	client/src/lang/zh_CN/users.json
#	client/src/lang/zh_HK/servers.json
#	client/src/lang/zh_HK/settings.json
#	client/src/lang/zh_HK/users.json
#	client/src/lang/zh_TW/servers.json
#	client/src/lang/zh_TW/settings.json
#	client/src/lang/zh_TW/users.json
#	client/src/main.js
#	client/src/plugins/api/index.js
#	client/src/plugins/api/servers.js
#	client/src/views/Account.vue
#	client/src/views/AddNode.vue
#	client/src/views/AddServer.vue
#	client/src/views/Login.vue
#	client/src/views/Node.vue
#	client/src/views/Servers.vue
#	client/src/views/Settings.vue
#	client/src/views/User.vue
#	client/src/views/Users.vue
#	cmd/migrate.go
#	database/loader.go
#	environments/docker/docker.go
#	go.mod
#	go.sum
#	legacy/panel.go
#	middleware/handlers/auth.go
#	middleware/handlers/oauth2.go
#	models/client.go
#	models/permission.go
#	models/user.go
#	models/usersetting.go
#	models/userview.go
#	services/oauth2.go
#	services/permission.go
#	services/token.go
#	services/user.go
#	services/usersettings.go
#	web/api/loader.go
#	web/api/self.go
#	web/api/servers.go
#	web/api/users.go
#	web/loader.go
#	web/oauth2/token.go
#	web/proxy/loader.go

disconsented committed:
Merge remote-tracking branch 'upstream/master' into feature/allow_duplicate_names

# Conflicts:
#	database/loader.go
#	database/loader_windows.go
#	go.mod
#	go.sum
#	models/client.go
#	services/oauth2.go

disconsented committed:
update go.mod

disconsented committed:
feat: Add additional migrations
